contracts/core/diaspore/cosigner/Collateral.sol

```diff
@@ -742,21 +756,10 @@ contract Collateral is Ownable, Cosigner, ERC721Base {
     function canWithdraw(
         uint256 _entryId,
         uint256 _debtInToken
-    ) public returns (int256) {
+    ) public view returns (int256) {
         Entry storage entry = entries[_entryId];
         // Valuate the entry amount in loanManagerToken
         uint256 collateralInToken = collateralToTokens(entry.token, entry.amount);
```

Comment:
RequestCosign and withdraw uses this function and can make a front running here
Because the function collateralToTokens takes the rate of the converter
Maybe it's collateralToTokens needs to take the rate from the the oracle of the entry to prevent the front runing